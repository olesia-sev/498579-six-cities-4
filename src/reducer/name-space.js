const NameSpace = {
  DATA: `DATA`,
  APP: `APP`,
  USER: `USER`,
};

export {NameSpace};
