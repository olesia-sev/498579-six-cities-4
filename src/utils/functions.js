export const getRandomNumberInRange = (min, max) => {
  return Math.floor(Math.random() * (max - min + 1)) + min;
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};
