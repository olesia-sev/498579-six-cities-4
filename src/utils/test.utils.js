export const optionsAmount = 100;

export const testString = `Test string`;

export const emptyFunction = () => {};

export const cardDataArray = [
  {
    id: 10,
    img: `https://placedog.net/260/200?id=2`,
    images: [
      `https://placedog.net/260/200?id=3`,
      `https://placedog.net/260/200?id=33`,
      `https://placedog.net/260/200?id=12`,
      `https://placedog.net/260/200?id=10`,
      `https://placedog.net/260/200?id=11`,
      `https://placedog.net/260/200?id=15`,
    ],
    price: 150,
    title: `Title`,
    placeType: `Flat`,
    numberOfBedrooms: 2,
    maxGuests: 3,
    rating: 4,
    saved: true,
    premium: true,
    featuresInside: [
      `Wi-Fi`, `Heating`,
    ],
    hostName: `Pete`,
    hostAvatar: `https://placedog.net/260/200?id=5`,
    userPro: true,
    description: `Description`,
  },
  {
    id: 100,
    img: `https://placedog.net/260/200?id=3`,
    images: [
      `https://placedog.net/260/200?id=33`,
      `https://placedog.net/260/200?id=99`,
      `https://placedog.net/260/200?id=89`,
      `https://placedog.net/260/200?id=29`,
      `https://placedog.net/260/200?id=13`,
      `https://placedog.net/260/200?id=16`,
    ],
    price: 150,
    title: `Title test`,
    placeType: `Flat`,
    numberOfBedrooms: 1,
    maxGuests: 2,
    rating: 4,
    saved: false,
    premium: false,
    featuresInside: [
      `Wi-Fi`, `Heating`,
    ],
    hostName: `Pete`,
    hostAvatar: `https://placedog.net/260/200?id=57`,
    userPro: false,
    description: `Description`,
  }
];
