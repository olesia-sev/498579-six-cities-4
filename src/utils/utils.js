export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const SORT_TYPES = {
  POPULAR: `Popular`,
  TO_HIGH_PRICE: `Price: low to high`,
  TO_LOW_PRICE: `Price: high to low`,
  TOP_RATED: `Top rated first`,
};

export const getSortedOffers = (offers, activeSort) => {
  switch (activeSort) {
    case SORT_TYPES.POPULAR:
      return offers.slice();
    case SORT_TYPES.TO_HIGH_PRICE:
      return offers.slice().sort((a, b) => a.price - b.price);
    case SORT_TYPES.TO_LOW_PRICE:
      return offers.slice().sort((a, b) => b.price - a.price);
    case SORT_TYPES.TOP_RATED:
      return offers.slice().sort((a, b) => b.rating - a.rating);
    default:
      return offers.slice();
  }
};

export const MAX_REVIEWS_LENGTH = 10;

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  FAVOURITES: `/favorites`,
  OFFER_DETAIL: `/offers/:id`,
};

export const RATING_STAR_WIDTH = 20;

export const MAX_IMAGES_AMOUNT = 6;
export const SLICE_BEGIN = 0;
