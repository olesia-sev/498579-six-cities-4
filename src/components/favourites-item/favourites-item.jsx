import React from 'react';
import PlaceCard from "../place-card/place-card";
import * as themes from "../place-card/place-card";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import {AppRoute} from "../../utils/utils";

const FavouritesItem = ({favourites, city}) => {
  return (
    <li key={city} className="favorites__locations-items">

      <div className="favorites__locations locations locations--current">
        <div className="locations__item">
          <Link to={AppRoute.ROOT} className="locations__item-link">
            <span>{city}</span>
          </Link>
        </div>
      </div>

      <div className="favorites__places">
        {
          favourites.filter((item) => item.cityId === city).map((favItem) => (
            <PlaceCard key={favItem.id} theme={themes.FAVOURITE_THEME} offer={favItem}/>
          ))
        }

      </div>
    </li>
  );
};

FavouritesItem.propTypes = {
  favourites: PropTypes.array.isRequired,
  city: PropTypes.string.isRequired,
};

export {FavouritesItem};
