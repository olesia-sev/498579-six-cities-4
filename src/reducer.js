import {extend} from './utils/functions';
import cities from "./mocks/cities";
import offers from "./mocks/offers";

export const initialState = {
  activeCityId: 10,
  cities,
  offers,
};

const ActionType = {
  SET_ACTIVE_CITY: `SET_ACTIVE_CITY`,
};

const ActionCreator = {
  setActiveCity: (id) => ({
    type: ActionType.SET_ACTIVE_CITY,
    payload: id,
  })
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_ACTIVE_CITY:
      return extend(state, {
        activeCityId: action.payload,
      });

    default: {
      return state;
    }
  }
};

export {reducer, ActionType, ActionCreator};
