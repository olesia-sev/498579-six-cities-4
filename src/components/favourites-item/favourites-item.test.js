import React from "react";
import renderer from "react-test-renderer";
import {FavouritesItem} from "./favourites-item";
import {cardDataArray} from "../../utils/test.utils";
import {BrowserRouter as Router} from "react-router-dom";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {NameSpace} from "../../reducer/name-space";


const mockStore = configureStore([]);

it(`FavouritesItem should be rendered`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      offers: cardDataArray,
    },
  });

  const tree = renderer
    .create(
        <Router>
          <Provider store={store}>
            <FavouritesItem favourites={cardDataArray} city={cardDataArray[0].cityId} />
          </Provider>
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
