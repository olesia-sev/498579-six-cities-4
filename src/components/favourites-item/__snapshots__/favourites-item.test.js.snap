// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouritesItem should be rendered 1`] = `
<li
  className="favorites__locations-items"
>
  <div
    className="favorites__locations locations locations--current"
  >
    <div
      className="locations__item"
    >
      <a
        className="locations__item-link"
        href="/"
        onClick={[Function]}
      >
        <span>
          Paris
        </span>
      </a>
    </div>
  </div>
  <div
    className="favorites__places"
  >
    <article
      className="favorites__card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="favorites__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offers/10"
          onClick={[Function]}
        >
          <img
            alt="Place image"
            className="place-card__image"
            height={110}
            src="https://placedog.net/260/200?id=2"
            width={150}
          />
        </a>
      </div>
      <div
        className="favorites__card-info place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              150
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button place-card__bookmark-button--active"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offers/10"
            onClick={[Function]}
          >
            Title
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Flat
        </p>
      </div>
    </article>
  </div>
</li>
`;
